src/pages/Feedback/Feedback.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect, useState } from 'react';
+import Rating from './Rating';
+import FormButton from '../../components/FormButton';
+import ReportForm from './ReportForm';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import { addRating, getMeeting } from '../../redux/slices/meetingSlice';
+
+const Feedback = () => {
+  const [showReport, setShowReport] = useState(false);
+  const [reportInput, setReportInput] = useState('');
+  const [starRating, setStarRating] = useState(0);
+  const [noRating, setNoRating] = useState(false);
+  const [noReportText, setNoReportText] = useState(false);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const meeting = useSelector((state) => state.meetings.meeting);
+  const user = useSelector((state) => state.auth.user);
+
+  const token = localStorage.getItem('token');
+  if (!token) {
+    navigate('/login');
+  }
+
+  const { meetingId } = useParams();
+
+  useEffect(() => {
+    async function fetchMeeting() {
+      const meetingFromDispatch = await dispatch(getMeeting({ meetingId }));
+      if (user.id) {
+        console.log('meeting.ratings', meetingFromDispatch.payload.ratings);
+
+        if (meetingFromDispatch.meta.requestStatus === 'fulfilled') {
+          if (
+            user.id &&
+            (meetingFromDispatch.payload.meetingStatus !== 'closed' ||
+              meetingFromDispatch.payload.isClosed ||
+              (user.id !== meetingFromDispatch.userId &&
+                user.id !== meetingFromDispatch.buddyId))
+          ) {
+            navigate('/');
+          }
+        }
+      }
+    }
+    fetchMeeting();
+  }, [user]);
+
+  async function submitRating(e) {
+    e.preventDefault();
+    if (starRating === 0) {
+      setNoRating(true);
+      return;
+    }
+    const createdRating = await dispatch(
+      addRating({ meetingId, newRating: { rating: starRating } })
+    );
+    if (createdRating.meta.requestStatus === 'rejected') {
+      alert(`Error when sending rating`);
+    } else if (createdRating.meta.requestStatus === 'fulfilled') {
+      navigate('/');
+    }
+  }
+
+  const submitReport = async (e) => {
+    e.preventDefault();
+    if (!reportInput) {
+      return setNoReportText(true);
+    }
+    const createdReport = await dispatch(
+      addRating({
+        meetingId,
+        newRating: { rating: 1, reportComment: reportInput, isReport: true },
+      })
+    );
+    if (createdReport.meta.requestStatus === 'rejected') {
+      alert(`Error when sending report`);
+    } else if (createdReport.meta.requestStatus === 'fulfilled') {
+      navigate('/');
+    }
+  };
+
+  const userReviews = meeting?.ratings?.reduce((acc, rating) => {
+    acc[rating.userId] = true;
+    return acc;
+  }, {});
+  if (userReviews && userReviews[user.id]) {
```

Comment:
We're checking that the user is not included in the list of reviews for this meeting - is that to imply that we're sending our buddy's review of ourself to the front end?